checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakTransfer.java

```diff
@@ -0,0 +1,276 @@
+package org.checkerframework.checker.resourceleak;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.checker.calledmethods.CalledMethodsTransfer;
+import org.checkerframework.checker.calledmethods.qual.EnsuresCalledMethodsVarArgs;
+import org.checkerframework.checker.mustcall.CreatesObligationElementSupplier;
+import org.checkerframework.checker.mustcall.MustCallAnnotatedTypeFactory;
+import org.checkerframework.checker.mustcall.MustCallChecker;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.dataflow.analysis.ConditionalTransferResult;
+import org.checkerframework.dataflow.analysis.TransferInput;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.block.ExceptionBlock;
+import org.checkerframework.dataflow.cfg.node.ArrayCreationNode;
+import org.checkerframework.dataflow.cfg.node.LocalVariableNode;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.dataflow.cfg.node.ObjectCreationNode;
+import org.checkerframework.dataflow.cfg.node.TernaryExpressionNode;
+import org.checkerframework.dataflow.expression.JavaExpression;
+import org.checkerframework.framework.flow.CFAbstractStore;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.TreeUtils;
+import org.checkerframework.javacutil.TypesUtils;
+
+/** The transfer function for the resource-leak extension to the called-methods type system. */
+public class ResourceLeakTransfer extends CalledMethodsTransfer {
+
+  /**
+   * {@link #makeExceptionalStores(MethodInvocationNode, TransferInput)} requires a TransferInput,
+   * but the actual exceptional stores need to be modified in {@link #accumulate(Node,
+   * TransferResult, String...)}, which only has access to a TransferResult. So this variable is set
+   * to non-null in {@link #visitMethodInvocation(MethodInvocationNode, TransferInput)} before the
+   * call to super, which will call accumulate(); this field is then reset to null afterwards to
+   * prevent it from being used somewhere it shouldn't be.
+   */
+  private @Nullable Map<TypeMirror, CFStore> exceptionalStores;
```

Comment:
@kelloggm should this logic around propagating called methods facts along exceptional edges be moved to the main Called Methods Checker?  I thought we agreed that in general, a method should be considered "called" even if the call results in an exception.

Reply:
https://github.com/typetools/checker-framework/pull/4683